test/rulesets/Base/arraymath.jl

```diff
@@ -6,26 +6,50 @@
         rrule_test(inv, randn(T, N, N), (B, randn(T, N, N)))
     end
 
-    @testset "*" begin
-        @testset "Matrix-Matrix" begin
-            dims = [3,4,5]
-            for n in dims, m in dims, p in dims
-                # don't need to test square case multiple times
-                n > 3 && n == m == p && continue
-                A = randn(m, n)
-                B = randn(n, p)
-                Ȳ = randn(m, p)
-                rrule_test(*, Ȳ, (A, randn(m, n)), (B, randn(n, p)))
-            end
+    # TODO Test on Complex once we have Complex support for all of these.
+    @testset "*: $T" for T in (Float64, ComplexF64)
+        ⋆(a) = round.(5*randn(T, a))  # Helper to generate nice random values
+        ⋆(a, b) = ⋆((a, b))  # matrix
+        ⋆() = only(⋆(()))  # scalar
+
+        ⋆₂(a) = (⋆(a), ⋆(a)) # Helper to generate random matrix and its cotangent
```

Comment:
What are these symbols? I don't think I've ever seen them before...

Reply:
`\star` and `\star\_2`

Reply:
Fair enough. Thanks!